### DISCUSSION TITLE
Generating Input Data

### DISCUSSION THREAD

**Discussion by npiroozan** [deepmodeling/deepmd-kit#1877] [Q&A]
Hello,

I hope you are doing well.  I'd like to ask a question regarding how to generate input data using Quantum Espresso:

What output is needed from QE in order to use dpdata and generate the input data for the DNN?

Right now from the QE I've used, the following files are outputted:

si.cel
si.pos
si.evp
(among a few more of course)

si.for  is currently empty which is a bit strange.  I'm using cp.x to run AIMD for Silicon.

Thank you for your advice!

**Comment by Yi-FanLi** [comment]
Hi Nariman,
To prepare the input data, you need at least the following 4 files:
si.in
si.pos
si.evp
si.for (if you want to use force during training).
I recommend you using `dpdata` to post-process your trajectory. You may refer to the following python snippet to process the data:
```
import numpy as np
import dpdata
d_log = dpdata.LabeledSystem('si', fmt = 'qe/cp/traj')
print(d_log)
idx = d_log.shuffle()
d_log_shuffled = dpdata.LabeledSystem()
for ii in idx:
  d_log_shuffled.append(d_log.sub_system(ii))
Nsets = 5
Neach = 200
for i in range(Nsets):
  print(d_log_shuffled.sub_system(np.arange(i*Neach, (i+1)*Neach).tolist()).get_nframes())
  d_log_shuffled.sub_system(np.arange(i*Neach, (i+1)*Neach).tolist()).to("deepmd/npy", "deepmd_data/"+str(i), set_size=Neach)
```
In this example I am preparing 5 sets with 200 frames in each set.

If you have empty `.for` file, could you please check if your `tprnfor` flag is set to be `TRUE`?

Thanks,
Yifan

**Reply by npiroozan** [reply]
Hi Yifan,

Thank you very much for the detailed response.  I very much appreciate it.  Indeed, you were right regarding tprnfor.  I forgot about that detail.  Thank you, sir.

One final question:

Just to confirm, "Si.in" refers to the input file?

Thank you very much.

**Reply by Yi-FanLi** [reply]
Yes, they refer to the input file.

**Comment by npiroozan** [comment]
I'm trying to generate input data for Silicon for a 432 atom system generated in the following paper:

Zhang, Linfeng, Jiequn Han, Han Wang, Wissam A. Saidi, Roberto Car, and Weinan E. "End-to-end Symmetry Preserving Inter-atomic Potential Energy Model for Finite and Extended Systems." arXiv preprint arXiv:1805.09003 (2018).

I'm trying to generate much larger input data for the same system (larger number of timesteps) in order to see if we can improve accuracy for Energy and Force (above 10^-4 for Energy and 10^-2 for Force).  I'm not familiar with Quantum Espresso.  Would it be possible to share the input script for the Silicon Run?  Or perhaps could you please review my input script.?  I want to make sure I'm doing it right. 

Thank you very much for your time.

**Reply by npiroozan** [reply]
My apologies.  I think a more appropriate question would be if there is a hard-coded limit to accuracy in DeePMD?

The best I have seen for Silicon is:

1x10^-4 for energy
1x10^-2 for force

How can one get better accuracy than this?

Thank you!

**Reply by njzjz** [reply]
You may firstly check if your data has enough precision.

**Reply by npiroozan** [reply]
Indeed, that is a good point.  To what degree is the accuracy of the model based on the amount of data?

Is there any benefit in using 1 million samples as opposed to 10 thousand?

**Reply by njzjz** [reply]
To be clear, you should check whether your energy data (`energy.npy` in NumPy format) has a precision of less than 1e-4.

For example, if the energy is something like 12345.6789 eV, of course, you can't expect it has a precision of 1e-5.

**Reply by npiroozan** [reply]
Understood, thank you very much.
